wgpu/src/renderer.rs

```diff
@@ -254,24 +254,31 @@ impl Renderer {
                 offset,
                 content,
             } => {
-                let clip_layer = Layer::new(
-                    Rectangle {
-                        x: bounds.x as u32 - layer.offset.x,
-                        y: bounds.y as u32 - layer.offset.y,
-                        width: bounds.width as u32,
-                        height: bounds.height as u32,
-                    },
-                    layer.offset + *offset,
-                );
-
-                let new_layer = Layer::new(layer.bounds, layer.offset);
-
-                layers.push(clip_layer);
-
-                // TODO: Primitive culling
-                self.draw_primitive(content, layers);
-
-                layers.push(new_layer);
+                let x = bounds.x - layer.offset.x as f32;
+                let y = bounds.y - layer.offset.y as f32;
+                let width = (bounds.width + x).min(bounds.width);
+                let height = (bounds.height + y).min(bounds.height);
+
+                // Only draw visible content on-screen
+                // TODO: Also, check for parent layer bounds to avoid further
```

Comment:
Is the`TODO` referring to a parent layer that is already defined/accessible?
or is it referring to part of #30 to be implemented and the `TODO` is a reminder for later?

Reply:
Yes, the parent layer should be the `layer` variable.

Implementing this `TODO` should be a simple rectangle intersection check. In fact, I think this is implementable for all primitives.

I have not decided to go for it yet because performance is not currently my main concern at this point. I prefer to keep code manageable and simple for now, while we are figuring things out.